auth/src/main/java/com/firebase/ui/auth/ui/idp/CredentialSignInHandler.java

```diff
@@ -82,9 +84,10 @@ public void onComplete(@NonNull Task<AuthResult> task) {
                             .addOnFailureListener(new OnFailureListener() {
                                 @Override
                                 public void onFailure(@NonNull Exception e) {
-                                    // TODO: What to do when signing in with Credential fails
-                                    // and we can't continue to Welcome back flow without
-                                    // knowing providers?
+                                    mHelper.finishActivity(
+                                            mActivity,
+                                            ResultCodes.CANCELED,
+                                            IdpResponse.getErrorCodeIntent(ErrorCodes.UNKNOWN_ERROR));
```

Comment:
@samtstern See my reply here: https://github.com/firebase/FirebaseUI-Android/issues/526#issuecomment-297532702. This would be a case where we could return the exception directly. However, with our current system in place, I agree with you that returning `UNKNOWN_ERROR` is the right thing to do in this case.